src/v/security/license.cc

```diff
@@ -0,0 +1,302 @@
+      });
+    if (schema_keys_found < v0_license_schema.size()) {
+        throw license_malformed_exception(
+          "Missing expected parameters from license");
+    }
+    lc.format_version = parse_int(doc.FindMember("version")->value);
+    if (lc.format_version < 0) {
+        throw license_invalid_exception("Invalid format_version, is < 0");
+    }
+
+    lc.organization = parse_str(doc.FindMember("org")->value);
+    if (lc.organization == "") {
+        throw license_invalid_exception("Cannot have empty string for org");
+    }
+    lc.type = integer_to_license_type(parse_int(doc.FindMember("type")->value));
+    lc.expiry = parse_expiry(doc.FindMember("expiry")->value);
+}
+
+license make_license(const ss::sstring& raw_license) {
+    license lc;
+    auto components = parse_license(raw_license);
+    if (!crypto::verify_license(components.data, components.signature)) {
+        throw license_verifcation_exception("RSA signature invalid");
+    }
+    try {
+        auto decoded_data = base64_to_string(components.data);
+        json::Document doc;
+        doc.Parse(decoded_data);
+        if (doc.HasParseError()) {
+            throw license_malformed_exception("Malformed data section");
+        }
+        parse_data_section(lc, doc);
+        return lc;
+    } catch (const base64_decoder_exception&) {
+        throw license_malformed_exception("Failed to decode data section");
+    }
+}
+
+bool license::is_expired() const noexcept {
+    return expiry < boost::gregorian::day_clock::universal_day();
+}
+
+long license::days_until_expires() const noexcept {
+    return is_expired()
+             ? -1
+             : (expiry - boost::gregorian::day_clock::universal_day()).days();
+}
+
+void license::serde_read(iobuf_parser& in, const serde::header& h) {
+    using serde::read_nested;
+    format_version = read_nested<uint8_t>(in, h._bytes_left_limit);
+    type = read_nested<license_type>(in, h._bytes_left_limit);
+    organization = read_nested<ss::sstring>(in, h._bytes_left_limit);
+    expiry = boost::gregorian::from_simple_string(
+      read_nested<ss::sstring>(in, h._bytes_left_limit));
+}
+
+void license::serde_write(iobuf& out) {
+    using serde::write;
+    write(out, format_version);
+    write(out, type);
+    write(out, organization);
+    write(
+      out,
+      ss::sstring(boost::gregorian::to_iso_extended_string_type<char>(expiry)));
+}
+
+} // namespace security
+
+namespace reflection {
+
+void adl<security::license>::to(iobuf& out, security::license&& l) {
+    vassert(true, "security::license should always use serde, never adl");
+    reflection::serialize(
+      out,
+      l.format_version,
+      l.type,
+      std::move(l.organization),
+      ss::sstring(
+        boost::gregorian::to_iso_extended_string_type<char>(l.expiry)));
+}
+
+security::license adl<security::license>::from(iobuf_parser& in) {
+    vassert(true, "security::license should always use serde, never adl");
```

Comment:
can we remove this now? or rather, why is adl defined at all here?

Reply:
@jcsp